src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs

```diff
@@ -442,6 +436,32 @@ public static DiagnosticData Create(Diagnostic diagnostic, TextDocument document
             return service?.GetAdditionalProperties(diagnostic);
         }
 
+        private static ImmutableArray<DiagnosticDataLocation> GetAdditionalLocations(TextDocument document, Diagnostic diagnostic)
+        {
+            if (diagnostic.AdditionalLocations.Count == 0)
+            {
+                return ImmutableArray<DiagnosticDataLocation>.Empty;
+            }
+
+            using var _ = ArrayBuilder<DiagnosticDataLocation>.GetInstance(diagnostic.AdditionalLocations.Count, out var builder);
+            foreach (var location in diagnostic.AdditionalLocations)
+            {
+                if (location.IsInSource)
+                {
+                    builder.AddIfNotNull(CreateLocation(document.Project.GetDocument(location.SourceTree), location));
+                }
+                else if (location.Kind == LocationKind.ExternalFile)
+                {
+                    if (document.Project.GetDocumentForExternalLocation(location) is { } textDocumentId)
```

Comment:
i don't think this check is needed.  GetTextDocument allows null doc-id, right?

Reply:
Thanks, done.